File: app/src/main/java/com/ahao/myapplication/banner/ScaleBannerLayoutManager.java
package com.ahao.myapplication.banner;

import android.view.View;

public class ScaleBannerLayoutManager extends BannerLayoutManager {
    private float heightScale = 0.9f;
    private float widthScale = 0.9f;

    protected void doWithItem() {
        if (heightScale >= 1 || widthScale >= 1) {
            return;
        }

        int childCount = getChildCount();

        for (int i = 0; i < childCount; i++) {
            View child = getChildAt(i);

            float itemMiddle = (getDecoratedRight(child) + getDecoratedLeft(child)) / 2.0f;
            float screenMiddle = mOrientationHelper.getTotalSpace() / 2.0f;
            float interval = Math.abs(screenMiddle - itemMiddle) * 1.0f;
            if (interval - 0f < 0.0001) {
                continue;
            }
            float ratio = 1 - (1 - heightScale) * (interval / itemWidth);
            float ratioWidth = 1 - (1 - widthScale) * (interval / itemWidth);
            child.setScaleX(ratioWidth);
            child.setScaleY(ratio);
        }
    }
}

File: bannerview/src/main/java/com/ahao/bannerview/ScaleBannerLayoutManager.kt
package com.ahao.bannerview

import kotlin.math.abs

class ScaleBannerLayoutManager : BannerLayoutManager() {

    private var heightScale: Float = 0.9f
    private var widthScale: Float = 0.9f

    override fun doWithItem() {
        if (heightScale >= 1 || widthScale >= 1) {
            return
        }

        for (i in 0 until childCount) {
            val child = getChildAt(i)
            val itemMiddle = (getDecoratedRight(child!!) + getDecoratedLeft(child)) / 2.0f
            val screenMiddle = mOrientationHelper.totalSpace / 2.0f
            val interval = abs(screenMiddle - itemMiddle) * 1.0f
            if (interval - 0f < 0.0001) {
                continue
            }
            val ratio = 1 - (1 - heightScale) * (interval / itemWidth)
            val ratioWidth = 1 - (1 - widthScale) * (interval / itemWidth)
            child.scaleX = ratioWidth
            child.scaleY = ratio
        }
    }

}